controllers/review.controller.js

```diff
@@ -0,0 +1,186 @@
+const { ObjectID } = require('mongodb');
+const Review = require('../models/review.model');
+const ReviewLib = require('../lib/review.lib');
+const asyncHandler = require('../middlewares/async.middleware');
+const ErrorResponse = require('../utils/errorResponse.util');
+const advancedResults = require('../utils/advancedResults.util');
+
+class ReviewController {
+  constructor() {
+    this.reviewLib = new ReviewLib();
+  }
+
+  /**
+   * @desc Create review
+   * @route POST /api/v1/reviews
+   * @access Private
+   */
+  postReview = asyncHandler(async (req, res) => {
+    // Add user to req.body
+    req.body.authorInformation = req.user;
+    const rawData = req.body;
+    await this.reviewLib.checkResourceInDB({ user: rawData.user, book: rawData.book });
+    const review = await this.reviewLib.createReview(rawData);
+    return res.status(201).json({
+      success: true,
+      data: review,
+    });
+  });
+
+  /**
+   * @desc Edit review
+   * @route PUT /api/v1/reviews/:id
+   * @access Private
+   */
+  putReview = asyncHandler(async (req, res, next) => {
+    // Add user to req.body
+    const { id } = req.params;
+    req.body.authorInformation = req.user;
+    const rawData = req.body;
+
+    let review = await this.reviewLib.fetchReview({ _id: id });
+    if (!review) {
+      return next(
+        new ErrorResponse(`Review with id: ${id} does not exist on the database`, 404),
+      );
+    }
+
+    review = await this.reviewLib.updateReview(id, rawData);
+    return res.status(202).json({
+      success: true,
+      data: review,
+    });
+  });
+
+  /**
+   * @desc Edit review to add a like
+   * @route PUT /api/v1/reviews/:id/likes
+   * @access Private
+   */
+  putReviewLike = asyncHandler(async (req, res, next) => {
+    // Add user to req.body
+    const { user, params, body } = req;
+    const { id } = params;
+    const { action } = body;
+
+    if (!['like', 'unlike'].includes(action)) {
+      return next(new ErrorResponse('Invalid action', 400));
+    }
+
+    let review = await this.reviewLib.fetchReview({ _id: id });
+    if (!review) {
+      return next(
+        new ErrorResponse(`Review with id: ${id} does not exist on the database`, 404),
+      );
```

Comment:
[Optional] 

`Review with id: ${id} does not exist` will suffice without `on the database`

Reply:
Okay